Compile error[E0004]: non-exhaustive patterns: `&Substring(_, _)` not covered

May be it is my fault not knowing were I can start a build:

Reproduce:
```
# git checkout 8d816a40 "This commit is the 364-credential-display commit, but I think that is not related."
# cd kanidm_unix_int
# cargo build --release
```

```
# ~/d/r/k/kanidm_unix_int ((8d816a40…))> cargo build --release
   Compiling arrayvec v0.5.2
   Compiling num-traits v0.2.14
   Compiling crossbeam-channel v0.5.0
   Compiling crossbeam-deque v0.8.0
   Compiling hyper-tls v0.5.0
   Compiling reqwest v0.11.2
   Compiling blake3 v0.3.7
   Compiling crossbeam v0.8.0
   Compiling num-integer v0.1.44
   Compiling num-complex v0.3.1
   Compiling chrono v0.4.19
   Compiling num-bigint v0.3.2
   Compiling num-iter v0.1.42
   Compiling async-session v2.0.1
   Compiling zxcvbn v2.1.1
   Compiling num-rational v0.3.2
   Compiling tide v0.15.1
   Compiling kanidm_proto v1.1.0-alpha.3 (/home/rene/dev/rust/kanidm/kanidm_proto)
   Compiling num v0.3.1
   Compiling tide-rustls v0.1.6
   Compiling kanidm_client v1.1.0-alpha.3 (/home/rene/dev/rust/kanidm/kanidm_client)
   Compiling concread v0.2.8
   Compiling kanidm v1.1.0-alpha.3 (/home/rene/dev/rust/kanidm/kanidmd)
error[E0004]: non-exhaustive patterns: `&Substring(_, _)` not covered
   --> kanidmd/src/lib/filter.rs:789:18
    |
789 |         Ok(match f {
    |                  ^ pattern `&Substring(_, _)` not covered
    |
   ::: /home/rene/.cargo/registry/src/github.com-1ecc6299db9ec823/ldap3_server-0.1.6/src/proto.rs:147:5
    |
147 |     Substring(String, LdapSubstringFilter),
    |     --------- not covered
    |
    = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
    = note: the matched value is of type `&ldap3_server::LdapFilter`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0004`.
error: could not compile `kanidm`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

# cargo version
cargo 1.50.0 (f04e7fab7 2021-02-04)
installed via rustup.

--- a/kanidmd/src/lib/filter.rs
+++ b/kanidmd/src/lib/filter.rs
@@ -20,7 +20,7 @@ use crate::value::{IndexType, PartialValue};
 use hashbrown::HashSet;
 use kanidm_proto::v1::Filter as ProtoFilter;
 use kanidm_proto::v1::{OperationError, SchemaError};
-use ldap3_server::simple::LdapFilter;
+use ldap3_server::proto::{LdapFilter, LdapSubstringFilter};
 // use smartstring::alias::String;
 use smartstring::alias::String as AttrString;
 use std::cmp::{Ordering, PartialOrd};
@@ -820,6 +820,18 @@ impl FilterComp {
                 FilterComp::Eq(a, v)
             }
             LdapFilter::Present(a) => FilterComp::Pres(ldap_attr_filter_map(a)),
+            LdapFilter::Substring(
+                _a,
+                LdapSubstringFilter {
+                    initial: _,
+                    any: _,
+                    final_: _,
+                },
+            ) => {
+                // let a = ldap_attr_filter_map(a);
+                ladmin_error!(audit, "Unable to convert ldapsubstringfilter to sub filter");
+                return Err(OperationError::FilterGeneration);
+            }
         })
     }
 }

